Cordate/Sources/Core/Calendar/CalendarDateSelectionController.swift

```diff
@@ -299,8 +299,20 @@ public class CalendarDateSelectionController: UIViewController {
             UIView.animate(withDuration: 0, animations: {
                 self.calendarLayout.setLayoutProperties(for: self.currentComponent)
                 self.calendarView.reloadData()
-                guard case .year = self.currentComponent, let year = self.selectedYear else { return }
-                if let indexPath = self.dataSource.indexOfYear(year).map({ IndexPath(item: $0, section: 0) }) {
+                guard case .year = self.currentComponent, let year = self.selectedYear else {
+                    switch self.currentComponent {
+                    case .year:
```

Comment:
I think `case .year where self.selectedYear == nil:` would eliminate one level of nesting here